src/Modules/UngradedModules/FreeMod.jl

```diff
@@ -308,3 +308,67 @@ end
 
 rels(F::FreeMod) = elem_type(F)[]
 
+function syzygy_generators(
+    g::Vector{T};
+    parent::Union{<:FreeMod, Nothing}=nothing
+  ) where {T<:FreeModElem}
+  isempty(g) && return Vector{T}()
+  F = Oscar.parent(first(g))
+  @assert all(Oscar.parent(x) === F for x in g) "parent mismatch"
+  R = base_ring(F)
+  m = length(g)
+  G = (parent === nothing ?  FreeMod(R, m) : parent)::typeof(F)
+  @assert ngens(G) == m "given parent does not have the correct number of generators"
```

Comment:
```suggestion
  @req ngens(G) != m "given parent does not have the correct number of generators"
```

Reply:
Do you happen to know by hard which type of error this throws so that I can adjust the tests?

Reply:
`ArgumentError`